pkgs/by-name/li/limine/package.nix

```diff
@@ -0,0 +1,59 @@
+# Builds limine with all available features.
+
+{
+  # Helpers
+  clangStdenv
+, fetchurl
+, lib
+, # Dependencies
+  llvmPackages
+, mtools
+, nasm
+}:
+
+let
+  version = "7.1.0";
+in
+clangStdenv.mkDerivation {
+  inherit version;
+  pname = "limine";
+  src = fetchurl {
+    url = "https://github.com/limine-bootloader/limine/releases/download/v${version}/limine-${version}.tar.gz";
+    sha256 = "sha256-oaQA461MGVuG7aJri2HyYb6Pu8BXOWTF3Wq/SputQSw=";
+  };
+
+  nativeBuildInputs = [
+    llvmPackages.bintools
+    llvmPackages.lld
+    mtools
+    nasm
+  ];
+
+  configureFlags = [
+    "TOOLCHAIN_FOR_TARGET=llvm"
+    "--enable-all"
+  ];
+
+  installFlags = [ "destdir=$out" "manprefix=/share" ];
+
+  outputs = [ "out" "doc" "dev" "man" ];
+
+  meta = with lib; {
+    homepage = "https://limine-bootloader.org/";
+    description = "Limine Bootloader";
+    # Caution. Some submodules have different licenses.
+    license = licenses.bsd2;
+    platforms = [
+      "i686-linux"
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-linux"
+      "aarch64-darwin"
+      "riscv-linux"
+    ];
```

Comment:
Isn't this just platforms.linux?

Reply:
I'm not exactly sure what to put here. I think, by using clang, this should (cross)compile on every architecture properly. However, the resulting files may not run on all platforms.

@48cf @mintsuki what can you say about the platform support of Limine?

Reply:
After an offline discussion with @blitz: We should use the intersection between platforms where Limine builds and where the userspace tools are useful. So I think

`platforms.linux` is correct. Possibly also `platforms.darwin`. Aye?